packages/widget_toolkit_pin/lib/src/lib_pin_code_with_biometrics/ui_components/pin_code_keyboard.dart

```diff
@@ -487,96 +554,271 @@ class _PinCodeKeyboardState extends State<PinCodeKeyboard>
               ),
               PinCodeKey(
                 number: 0,
-                isLoading: widget.isLoading,
+                isLoading: isLoading,
                 onPressed: _onKeyPressed,
               ),
-              SizedBox(
-                height: calculateKeyboardButtonSize(context),
-                width: calculateKeyboardButtonSize(context),
-                child: Center(
-                  child:
-                      widget.bottomRightKeyboardButton ?? _buildApplyButton(),
-                ),
-              ),
+              _buildBiometricsButton(context, isPinCodeIsSecureStorage,
+                  hasFingerScan, hasFaceScan, biometricsEnabled),
             ],
           ),
         ],
       );
 
-  Widget _buildApplyButton() => AnimatedSwitcher(
+  Widget _buildBiometricsButton(
```

Comment:
I believe the builder Helper methods can be reduced/avoided

Reply:
I have deleted some of them, but I think those `_buildIconContent()` `_buildButtonContent` should be left as separate methods and with if() statements instead of with Ternary operator for easier debugging if this is needed.